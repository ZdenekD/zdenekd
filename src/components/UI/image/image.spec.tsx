import {shallow} from 'enzyme';
import findComponent from '../../../__test__/utils/helpers';
import Image from '.';

describe('Image', () => {
    it('match snapshot', () => {
        const wrapper = shallow(<Image src="/path/to/image.jpg" type="webp, avif" />);

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const wrapper = shallow(<Image src="/path/to/image.jpg" />);
        const component = findComponent(wrapper, 'component-image');

        expect(component.exists()).toBe(true);
    });

    it('renders `webp` source', () => {
        const wrapper = shallow(<Image src="/path/to/image.jpg" type="webp" />);
        const component = findComponent(wrapper, 'component-image-webp');

        expect(component.exists()).toBe(true);
    });

    it('renders `avif` source', () => {
        const wrapper = shallow(<Image src="/path/to/image.jpg" type="avif" />);
        const component = findComponent(wrapper, 'component-image-avif');

        expect(component.exists()).toBe(true);
    });
});
