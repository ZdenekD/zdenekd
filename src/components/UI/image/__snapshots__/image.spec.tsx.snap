// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image match snapshot 1`] = `
<picture class="picture"
         data-test="component-image"
>
  <source srcset="/path/to/image.avif"
          type="image/avif"
          data-test="component-image-avif"
  >
  <source srcset="/path/to/image.webp"
          type="image/webp"
          data-test="component-image-webp"
  >
  <img src="/path/to/image.jpg"
       alt
       loading="lazy"
       decoding="async"
       class="image "
  >
</picture>
`;
